x-pack/plugins/ml/public/jobs/jobs_list_new/components/jobs_list/job_description.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const COLORS = [
+  '#00B3A4', // euiColorVis0
+  '#3185FC', // euiColorVis1
+  '#DB1374', // euiColorVis2
+  '#490092', // euiColorVis3
+  // '#FEB6DB', // euiColorVis4 light pink, too hard to read with white text
+  '#E6C220', // euiColorVis5
+  '#BFA180', // euiColorVis6
+  '#F98510', // euiColorVis7
+  '#461A0A', // euiColorVis8
+  '#920000', // euiColorVis9
+
+  '#666666', // euiColorDarkShade
+  '#0079A5', // euiColorPrimary
+];
+
+const colorMap = {};
+
+export function JobDescription({ job }) {
+  return (
+    <React.Fragment>
+      <div className="job-description">{job.description} &nbsp;
+        {
+          job.groups.map(group => (
+            <JobGroup key={group} name={group} />
+          ))
+        }
+      </div>
+    </React.Fragment>
+  );
+}
+JobDescription.propTypes = {
+  job: PropTypes.object.isRequired,
+};
+
+function JobGroup({ name }) {
+  return (
+    <div
+      className="inline-group"
+      style={{ backgroundColor: tabColor(name) }}
+    >
+      {name}
+    </div>
+  );
+}
+JobGroup.propTypes = {
+  name: PropTypes.string.isRequired,
+};
+
+function tabColor(name) {
+  if (colorMap[name] === undefined) {
+    const n = stringHash(name);
```

Comment:
This could use a comment, to say we do a hash on the group name to get the name mapping to the same colour in subsequent renders.